src/ipc.c

```diff
@@ -183,6 +183,40 @@ static Window *visible_windows_on_curr_output(i3ipcCon *root)
     return visible_windows(con);
 }
 
+static gint compare_rects(i3ipcRect *a, i3ipcRect *b)
+{
+    return a->y == b->y ? a->x - b->x : a->y - b->y;
+}
+
+static gint compare_workspace_position(gconstpointer a, gconstpointer b, gpointer outputs_)
+{
+    const i3ipcWorkspaceReply *reply_a = (i3ipcWorkspaceReply *) a;
+    const i3ipcWorkspaceReply *reply_b = (i3ipcWorkspaceReply *) b;
+
+    if (g_str_equal(reply_a->output, reply_b->output))
+        return 0;
+
+    i3ipcRect *rect_a = NULL;
+    i3ipcRect *rect_b = NULL;
+    GSList *outputs = (GSList *) outputs_;
+    const GSList *output_reply;
+    for (output_reply = outputs; output_reply; output_reply = output_reply->next)
+    {
+        i3ipcOutputReply *output = output_reply->data;
+        if (!output->active)
```

Comment:
Do we know whether the workspace reply only contains workspaces on active outputs?

Reply:
Non active outputs shouldn't have assigned workspaces:
https://github.com/i3/i3/blob/7411dfa5efc0c6035e7c644cd9c64166f516e81a/src/randr.c#L936-L940

No reason to risk a segfault assuming no bugs in i3 though. It's not like there are huge benefits skipping them.

Reply:
Good, I was more thinking whether we need to check `rect_a` and `rect_b` for `NULL` later

Reply:
I have thought about it. They could be `NULL` in the rare case where the program is called while the output configuration changes.

Reply:
Yeah, I added this. I almost didn't want to since the chances to happen are astronomically low.